lib/matplotlib/contour.pyi

```diff
@@ -103,12 +103,17 @@ class ContourSet(ContourLabeler, Collection):
     clip_path: Patch | Path | TransformedPath | TransformedPatchPath | None
     labelTexts: list[Text]
     labelCValues: list[ColorType]
-    allkinds: list[np.ndarray]
-    tcolors: list[tuple[float, float, float, float]]
+    @property
+    def tcolors(self) -> list[tuple[float, float, float, float]]: ...
 
     # only for not filled
-    tlinewidths: list[tuple[float]]
+    @property
+    def tlinewidths(self) -> list[tuple[float]]: ...
 
+    @property
+    def allkinds(self) -> list[np.ndarray]: ...
+    @property
+    def allsegs(self) -> list[np.ndarray]: ...
```

Comment:
If I've read these correctly, these are lists of lists of arrays:
```suggestion
    def allkinds(self) -> list[list[np.ndarray]]: ...
    @property
    def allsegs(self) -> list[list[np.ndarray]]: ...
```

Reply:
Yes, and technically the source of these is `Path.codes` and `Path.vertices`, which actually are only typed as `ArrayLike`... and the former can actually just be `None` (though that is not reflected by the type hint) as well (which I saw when I just ran it to be sure it was actually nested lists)

It is one of these "in practice in this usage the type _will_ be ndarray or None", but that is relying on knowledge outside of the type system... so I suspect if we were to inline the type hints it would cause a clash...

I also think that the `None` behavior is not super intentional in this context... There is a note in the `Path` docstring that says "don't use codes directly, use `iter_segments` so that you don't get the Nones from codes"

Reply:
Only the empty path I _think_ will _actually_ have `codes=None`... but that is in fact a case covered by the code...